scripts/vote_shapella.py

```diff
@@ -0,0 +1,212 @@
+    template = ShapellaUpgradeTemplate.at(template_address)
+    return template.address, template.finishUpgrade.encode_input()
+
+
+def start_vote(
+    tx_params: Dict[str, str], silent: bool, template_address: str
+) -> Tuple[int, Optional[TransactionReceipt]]:
+    """Prepare and run voting."""
+    upgrade_config = load_shapella_deploy_config()
+
+    # TODO: add upgrade_ prefix to the voting script to enable snapshot tests
+
+    voting: interface.Voting = contracts.voting
+    lido: interface.Lido = contracts.lido
+    lido_oracle: interface.LegacyOracle = contracts.legacy_oracle
+    node_operators_registry: interface.NodeOperatorsRegistry = contracts.node_operators_registry
+    staking_router = upgrade_config["stakingRouter"]["address"]
+
+    call_script_items = [
+        # TODO
+        encode_template_start_upgrade(template_address),
+        # 1. Publishing new implementation(TODO)
+        #                   in Lido app APM repo 0xF5Dc67E54FC96F993CD06073f71ca732C1E654B1
+        add_implementation_to_lido_app_repo(
+            update_lido_app["version"], update_lido_app["new_address"], update_lido_app["content_uri"]
+        ),
+        # 2. Updating implementation of Lido app with the new one TODO
+        update_app_implementation(update_lido_app["id"], update_lido_app["new_address"]),
+        # 3. Publishing new implementation (TODO)
+        #                   in Node Operators Registry app APM repo 0x0D97E876ad14DB2b183CFeEB8aa1A5C788eB1831
+        add_implementation_to_nos_app_repo(
+            update_nos_app["version"], update_nos_app["new_address"], update_nos_app["content_uri"]
+        ),
+        # 4. Updating implementation of Node Operators Registry app
+        #                   with the new one TODO
+        update_app_implementation(update_nos_app["id"], update_nos_app["new_address"]),
+        # 5. Publishing new implementation (TODO)
+        #     in Oracle app APM repo 0xF9339DE629973c60c4d2b76749c81E6F40960E3A
+        add_implementation_to_oracle_app_repo(
+            update_oracle_app["version"], update_oracle_app["new_address"], update_oracle_app["content_uri"]
+        ),
+        # 6. Updating implementation of Oracle app with new one TODO
+        update_app_implementation(update_oracle_app["id"], update_oracle_app["new_address"]),
+        # 7. Create permission for SET_EL_REWARDS_VAULT_ROLE of Lido app
+        #    assigning it to Voting 0x2e59A20f205bB85a89C53f1936454680651E618e
+        encode_permission_create(
+            entity=staking_router,
+            target_app=node_operators_registry,
+            permission_name="STAKING_ROUTER_ROLE",
+            manager=voting,
+        ),
+        # 8. Finalize upgrade via template
+        encode_template_finish_upgrade(template_address),
+        # 9+. Revoke obsolete roles
+        # TODO: on goerli the list is larger
+        encode_permission_revoke(lido, "MANAGE_FEE", revoke_from=voting),
```

Comment:
decided to extract into the separate vote and add a special helper in the Template contract